sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/CliSuite.scala

```diff
@@ -573,4 +573,53 @@ class CliSuite extends SparkFunSuite with BeforeAndAfterAll with Logging {
     // the date formatter for `java.sql.LocalDate` must output negative years with sign.
     runCliWithin(1.minute)("SELECT MAKE_DATE(-44, 3, 15);" -> "-0044-03-15")
   }
+
+  test("SPARK-33100: Ignore a semicolon inside a bracketed comment in spark-sql") {
+    runCliWithin(1.minute)("/* SELECT 'test';*/ SELECT 'test';" -> "test" )
+    runCliWithin(1.minute)(";;/* SELECT 'test';*/ SELECT 'test';" -> "test" )
+    runCliWithin(1.minute)("/* SELECT 'test';*/;; SELECT 'test';" -> "test" )
+    runCliWithin(1.minute)("SELECT 'test'; -- SELECT 'test';" -> "")
+    runCliWithin(1.minute)("SELECT 'test'; /* SELECT 'test';*/" -> "")
+    runCliWithin(1.minute)("/*$meta chars{^\\;}*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("/*\nmulti-line\n*/ SELECT 'test';" -> "test")
+    runCliWithin(1.minute)("/*/* multi-level bracketed*/ SELECT 'test';" -> "test")
+
+    val dataFilePath =
+      Thread.currentThread().getContextClassLoader.getResource("data/files/small_kv.txt")
```

Comment:
Could you make  a new test unit for this test?

Reply:
How about the case below? 
```
select /*    + broadcast(t1) */ * from t1 join t2 on t1.id = t2.id;
         ^^^
         space, indent, ...
```

Reply:
sure, will try it

Reply:
Have you added tests for the case above?

Reply:
sorry, I will add a test for this case.

Reply:
plan for  `select /*    + broadcast(t) */ * from t`

```
== Parsed Logical Plan ==
'Project [*]
+- 'UnresolvedRelation [t], [], false

== Analyzed Logical Plan ==
k: int, v: int
Project [k#218, v#219]
+- SubqueryAlias t
   +- Project [k#218, v#219]
      +- SubqueryAlias t
         +- LocalRelation [k#218, v#219]

== Optimized Logical Plan ==
LocalRelation [k#218, v#219]

== Physical Plan ==
LocalTableScan [k#218, v#219]
```


I do not find the way to match result for this case.
Because the `captureOutput` method capture output from source `output` one line by one line.

It can not match two  continue lines.

Reply:
will try to use `MERGEJOIN(t)` to verify broadcastjoin or mergejoin